inc/infrastructure/class-plugin.php

```diff
@@ -0,0 +1,103 @@
+<?php
```

Comment:
This is a nit and a general style comment and doesn't really block anything _but_ is this file _doing_ anything that isn't or couldn't be done elsewhere? It feels like a lot of setup stuff that could just be done in the namespaced procedural `namespace.php` file. The class structure here and in `class-service.php` feels like premature overoptimization. Prove me wrong?

I don't have a strong opinion about refactoring/removing this structure, I just want to understand why it's necessary.

Reply:
Discussed in slack/zoom. See [8c55797](https://github.com/pantheon-systems/pantheon-wordpress-edge-integrations/pull/8/commits/8c55797130e795f36944495e55f79d439f350466).